### PR TITLE
Refactor tests dst indentation

### LINKED ISSUE #1302: Update `openfga/language` Go module for DSL transformer/parser and refactor usages
### Checklist

- [X] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Describe the problem you'd like to have solved

A lot of our tests today use the openfga/language parser for Go to make it easier to write tests using our DSL syntax instead of the JSON model syntax. You can see a few examples of this here:

https://github.com/openfga/openfga/blob/main/internal/graph/check_test.go#L43-L55
https://github.com/openfga/openfga/blob/main/internal/graph/cached_resolver_test.go#L449-L469
https://github.com/openfga/openfga/blob/main/pkg/server/server_test.go#L182-L188

A more recent commit/release of the parser supports whitespace better, and so we can refactor these usages to have better usage of whitespace in the test files. The existing tests today had to swallow whitespace to avoid parser errors, and so they read quite cryptically relative to the other whitespace in the file.

### Describe the ideal solution

Upgrade the parser and fix any of the whitespace in the tests that use the parser to parse a DSL model string.

### Alternatives and current workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 00chorch**
Fixes #1302 for tests in "pkg/server" and "internal"

## Description

Now that openfga/language parser has better support for whitespaces, refactor to
have better readability in the test files.

## References

Issue #1302 

## Review Checklist
- [X] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [X] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: internal/condition/eval/eval_test.go [hunk 1]
```diff
@@ -27,53 +27,56 @@ func TestEvaluateTupleCondition(t *testing.T) {
...
-type document
-  relations
-    define can_view: [user with correct_ip]
-
-condition correct_ip(ip: string) {
-	ip == "192.168.0.1"
-}`),
+			model: parser.MustTransformDSLToProto(`
+				model
+				schema 1.1
```
Comment (adriantam) [inline internal/condition/eval/eval_test.go:32]: should schema be indented?

Comment (dev-chorch) [reply internal/condition/eval/eval_test.go:32]: Good catch Adrian, based on a [comment](https://github.com/openfga/openfga/pull/1597#discussion_r1597522757) on closed PR, that was desired.

I'll re-run script to incorporate that and re-submit today.

Comment (00chorch) [reply internal/condition/eval/eval_test.go:32]: Good catch Adrian, based on a [comment](https://github.com/openfga/openfga/pull/1597#discussion_r1597522757) on closed PR, that was desired.

I'll re-run script to incorporate that and re-submit today.

**Comment by 00chorch** [comment]
Hi @adriantam / Team, 
this PR remains in draft as I'm not sure how to fix the benchmark test (list_users.go) failing on line 186.

https://github.com/openfga/openfga/blob/c249beadaa519ca34fdb4b1ce81e1787c512c5a4/pkg/server/test/list_users.go#L180-L186

Reproduced locally using upstream/main branch, so I'm assuming it will be spot/fixed on main in the short term. As soon as I see a fix I will merge it and move this PR back to Review Ready state

**Comment by rhamzeh** [comment]
Thanks for the PR @00chorch !

Someone from the team will be looking at this shortly

**Review by jpadilla — APPROVED** [review]

**Review by jpadilla — APPROVED** [review]

**Pull request merged**